<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>React expects a unique identifier for performance optimizations. An array index is not a stable identifier most of the time. This results in unnecessary renders when the array items change index following some mutation. When components have state, this might also provoke bugs that are hard to diagnose.</p>
</div>
<div class="paragraph">
<p>We recommend using an explicit identifier to avoid misuse and accidental re-renders. If there is no unique attribute available, consider concatenating existing properties - hashing them if necessary - or creating a dedicated unique identifier.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function generateButtons(props) {
  return props.buttons.map((button, index) =&gt; {
    &lt;Button key={index}&gt;{button.number}&lt;/Button&gt;
  });
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function generateButtons(props) {
  return props.buttons.map((button, index) =&gt; {
    &lt;Button key={button.number}&gt;{button.number}&lt;/Button&gt;
  });
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://reactjs.org/docs/reconciliation.html#recursing-on-children">Recursing On Children</a> - React API reference</p>
</li>
<li>
<p><a data-rspec-id="S6477" class="rspec-auto-link">S6477</a></p>
</li>
<li>
<p><a data-rspec-id="S6486" class="rspec-auto-link">S6486</a></p>
</li>
</ul>
</div>
</div>
</div>