<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Disallow the use of optional chaining in an expression where the <code>undefined</code> value would raise an error.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The optional chaining operator <code>?.</code> allows to access a deeply nested property, returning <code>undefined</code> if the property or any intermediate object is <code>undefined</code>.</p>
</div>
<div class="paragraph">
<p>This usually means that we expect the expression to evaluate as <code>undefined</code> in some cases. Therefore, using the optional chaining operator in a context where returning <code>undefined</code> is forbidden can lead to errors.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Since optional chaining represents multiple execution branches, having an error thrown in such a context can be hard to debug.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You should provide fallbacks for when the optional chaining operator is used to avoid runtime errors.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">new (foo?.bar)();
const { foo } = bar?.baz;
const foo = [...bar?.baz]</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">new (foo?.bar ?? baz)()
const { foo } = bar?.baz || {}
const foo = bar?.baz ? [...bar.baz] : []</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>