<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Regular expressions have their own syntax that is understood by regular expression engines. Those engines will throw an exception at runtime if they are given a regular expression that does not conform to that syntax.</p>
</div>
<div class="paragraph">
<p>To avoid syntax errors, special characters should be escaped with backslashes when they are intended to be matched literally and references to capturing groups should use the correctly spelled name or number of the group.</p>
</div>
<div class="paragraph">
<p>Negative lookahead and negative lookbehind groups cannot be combined with <code>RegexOptions.NonBacktracking</code>. Such combination would throw an exception during runtime.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(Input As String)
    Dim Rx As New Regex("[A")                   ' Noncompliant
    Dim Match = Regex.Match(Input, "[A")        ' Noncompliant
    Dim Matches = Regex.Matches(Input, "[A")    ' Noncompliant
    Dim Replace = Regex.Replace(Input, "[A", "replacement")     ' Noncompliant
    Dim Split = Regex.Split(Input, "[A")        ' Noncompliant
    If (Regex.IsMatch(Input, "[A")) Then        ' Noncompliant
    End If
    Dim NegativeLookahead As New Regex("a(?!b)", RegexOptions.NonBacktracking)      ' Noncompliant
    Dim NegativeLookbehind As New Regex("(?&lt;!a)b", RegexOptions.NonBacktracking)    ' Noncompliant
End Sub</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">Public Sub DoSomething(Input As String)
    Dim Rx As New Regex("[A-Z]")
    Dim Match = Regex.Match(Input, "[A-Z]")
    Dim Matches = Regex.Matches(Input, "[A-Z]")
    Dim Replace = Regex.Replace(Input, "[A-Z]", "replacement")
    Dim Split = Regex.Split(Input, "[A-Z]")
    If (Regex.IsMatch(Input, "[A-Z]")) Then
    End If
    Dim NegativeLookahead As New Regex("a(?!b)")
    Dim NegativeLookbehind As New Regex("(?&lt;!a)b")
End Sub</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Fix the syntax error inside this regex</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The error inside the regular expression</p>
</div>
</div>
</div>
</div>