<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The PHP 4 method of declaring a variable, using the <code>var</code> keyword, was deprecated in early versions of PHP 5. Even though it&#8217;s not considered deprecated in the most recent versions, it&#8217;s nonetheless not best practice to use it. When <code>var</code> does appear, it is interpreted as a synonym for <code>public</code> and treated as such. Therefore <code>public</code> should be used instead.</p>
</div>
<div class="paragraph">
<p>From the PHP Manual:</p>
</div>
<div class="quoteblock">
<blockquote>
<div class="paragraph">
<p>The PHP 4 method of declaring a variable with the var keyword is still supported for compatibility reasons (as a synonym for the public keyword). In PHP 5 before 5.1.3, its usage would generate an E_STRICT warning.</p>
</div>
</blockquote>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php
class Foo
{
    var $bar = 1;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">&lt;?php
class Foo
{
    public $bar = 1;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Replace the "var" keyword with the modifier "public".</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_27_may_2014_200153_ann_campbell_wrote">on 27 May 2014, 20:01:53 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>The reference I found has a slightly different reading on the deprecation: http://www.php.net/manual/en/language.oop5.properties.php</p>
</div>
</div>
</div>
</div>