<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Nested components are defined in an enclosing one and can be simple functions or arrow expressions. React recreates them systematically during the render pass because it doesn&#8217;t know they haven&#8217;t changed. This hurts performance as the components are recreated too many times. However, this can also hide surprising bugs where the state of the nested components is lost between renders. Trying to use <code>useCallback</code> hooks for child components is also discouraged. You should actually refactor this code to define a component on its own, passing props if needed.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function Component() {
  function UnstableNestedComponent() {
    return &lt;div /&gt;;
  }

  return (
    &lt;div&gt;
      &lt;UnstableNestedComponent /&gt;
    &lt;/div&gt;
  );
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function SomeComponent({ footer: Footer }) {
  return (
    &lt;div&gt;
      &lt;Footer /&gt;
    &lt;/div&gt;
  );
}

function Component() {
  return (
    &lt;div&gt;
      &lt;SomeComponent footer={ () =&gt; &lt;div /&gt; } /&gt; { /* footer is a component nested inside 'Component' */ }
    &lt;/div&gt;
  );
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Component extends React.Component {
  render() {
    function UnstableNestedComponent() {
      return &lt;div /&gt;;
    }

    return (
      &lt;div&gt;
        &lt;UnstableNestedComponent /&gt;
      &lt;/div&gt;
    );
  }
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function OutsideDefinedComponent(props) {
  return &lt;div /&gt;;
}

function Component() {
  return (
    &lt;div&gt;
      &lt;OutsideDefinedComponent /&gt;
    &lt;/div&gt;
  );
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function Component() {
  return &lt;SomeComponent footer={ &lt;div /&gt; } /&gt;;
}</code></pre>
</div>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">class Component extends React.Component {
  render() {
    return (
      &lt;div&gt;
        &lt;SomeComponent /&gt;
      &lt;/div&gt;
    );
  }
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://reactjs.org/docs/reconciliation.html#elements-of-different-types">Elements Of Different Types</a> - React documentation</p>
</li>
</ul>
</div>
</div>
</div>