<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no point in creating objects that don&#8217;t effect program outcome. The creation of such objects is likely to indicate an error. Perhaps an assignment was forgotten or a method left uncalled?</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when an object is created and its state is updated, but it neither leaves the method where it was created nor affects the enclosing object.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void addPerson(String name){
  Person p = new Person();  // Noncompliant
  p.gender = Person.Gender.MALE;
  p.setName(name);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void addPerson(){
  Person p = new Person();
  p.gender = Person.Gender.MALE;
  p.setName("Sam");

  people.add(p);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this object; its creation has no effect on the overall program execution.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s3063">is related to: <a data-rspec-id="S3063" class="rspec-auto-link">S3063</a></h3>

</div>
<div class="sect2">
<h3 id="_on_28_jul_2015_075454_nicolas_peru_wrote">on 28 Jul 2015, 07:54:54 Nicolas Peru wrote:</h3>
<div class="literalblock">
<div class="content">
<pre>this rule is redundant with RSPEC-1848 which should be extended with this case rather than creating a new rule.</pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_on_14_aug_2015_121446_ann_campbell_wrote">on 14 Aug 2015, 12:14:46 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>\[~nicolas.peru] please create a POC of merging this with RPSEC-1848 to make sure it doesn&#8217;t create too many FP&#8217;s</p>
</div>
</div>
</div>
</div>