<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Certain mathematical comparisons will always return the same value, and should simply not be made.</p>
</div>
<div class="paragraph">
<p>These comparisons will return either always <code>true</code> or always <code>false</code> depending on the kind of comparison:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Comparing a <code>char</code> with a numeric constant that is outside of the range of <code>char</code>.</p>
</li>
<li>
<p>Comparing a <code>float</code> with a numeric constant that is outside of the range of <code>float</code>.</p>
</li>
<li>
<p>Comparing a <code>long</code> with a numeric constant that is outside of the range of <code>long</code>.</p>
</li>
<li>
<p>Comparing a <code>ulong</code> with a numeric constant that is outside of the range of <code>ulong</code>.</p>
</li>
</ul>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">float f = 42.0f;
if (f &lt;= double.MaxValue) { } // Noncompliant</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this comparison; it will always return (false|true).</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_31_jan_2023_155454_grigorios_paidis_wrote">on 31 Jan 2023, 15:54:54 Grigorios Paidis wrote:</h3>
<div class="paragraph">
<p>The rule is complementary to <a href="https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0652?f1url=%3FappId%3Droslyn%26k%3Dk(CS0652)">CS0652</a>. Thus, it only extends the functionality of this Compiler Warning.</p>
</div>
</div>
</div>
</div>