<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Disallow assigning values to variables used for imports.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Assigning a new value to an import variable is a poor practice that affects code readability.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>Bad practices can cause bugs. When using the reassigned variable, if you are expecting to access the imported value, but are in fact using something else can lead to unexpected behaviour that is hard to debug.</p>
</div>
<div class="paragraph">
<p>More importantly, it will cause a runtime error when using ES modules. With TypeScript, it will raise a compilation error.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>You should use a separate variable to store your other values and make constant the import variable.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">let { readFile } = require('fs');
readFile = 'hello world!';</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const { readFile } = require('fs');
const text = 'hello world!';</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>