<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Theoretically, any string of characters can form a pattern, and therefore a regular expression. However, it is possible to string together a set of characters which do not form a valid regular expression because they use unmatched, and unescaped characters which have special meaning in a regular expression context.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public boolean matchesPattern(String candidate) {
  String pattern = "a(b";  // this is fine as a plain string
  return candidate.matches("a(b");  // Noncompliant; was "a(b)" or "a\\(b" meant instead?
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>"xxx" isn&#8217;t a valid regular expression.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2639">is related to: <a data-rspec-id="S2639" class="rspec-auto-link">S2639</a></h3>

</div>
</div>
</div>