<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Declaring that a method <code>throws</code> an exception type that it doesn&#8217;t is like a wart on the code: seemingly ugly but harmless. Unfortunately, like warts, those un-thrown exceptions will spread as each calling method is forced to deal with the <code>throws</code> declaration. Eventually, your code is so covered in warts that you can&#8217;t see the form beneath.</p>
</div>
<div class="paragraph">
<p>Which is why this rule raises an issue when a method declares an exception type it doesn&#8217;t actually throw.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void output(String text) throws NumberFormatException {  // Noncompliant
  System.out.println(text);
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void output(String text) {
  System.out.println(text);
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the "throws" declaration for "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>unthrown exception type</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2018_100427_amaury_levé_wrote">on 9 Jan 2018, 10:04:27 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>This rule is irrelevant for C# and VB.Net as we do not declare the exceptions that can be thrown.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_23_jan_2019_103839_loïc_joly_wrote">on 23 Jan 2019, 10:38:39 Loïc Joly wrote:</h3>
<div class="paragraph">
<p>Irrelevant for C&#43;&#43; too, where throw specifications are removed&#8230;&#8203;</p>
</div>
</div>
</div>
</div>