<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>It can be useful to use the in-code comment <code>NOSONAR</code> to suppress issues on given lines, but when those suppressions are no longer relevant they become a potential source of confusion and should be removed.</p>
</div>
<div class="paragraph">
<p>This rule raises an issue when <code>NOSONAR</code> is used on a line for which no issues would be raised during analysis.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class A {
  // This constant definition is valid and not raising any issue
  static final int LOCAL = 42;  // NOSONAR - Noncompliant</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public class A {
  // This constant definition is valid and not raising any issue
  static final int LOCAL = 42;</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove the NOSONAR comment, no issue is raised in the related block</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>the NOSONAR comment</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s4290">relates to: <a data-rspec-id="S4290" class="rspec-auto-link">S4290</a></h3>

</div>
</div>
</div>