include::../description.adoc[]

== Noncompliant Code Example

[source,csharp]
----
public int Transform(int value)
{
    if (value <= 0)
    {
        return value;
    }
    int number = int.MaxValue;
    return number + value;  // Noncompliant
}
----

== Compliant Solution

[source,csharp]
----
public long Transform(int value)
{
    if (value <= 0)
    {
        return value;
    }
    long number = int.MaxValue;
    return number + value;
}
----

include::../rspecator.adoc[]
