<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>PHP doesn&#8217;t fail if two <code>define</code> statements assign a value to the same constant name: in such a case, PHP only issues a warning and ignores the second <code>define</code>. However, that code should be fixed. In the best case, the second <code>define</code> tries to assign the same value to the constant: it&#8217;s just useless code which can be removed. In the worst case, the second <code>define</code> attempts to assign a different value and is followed by code which wrongly assumes the value of the constant: it can be a nasty bug.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-php" data-lang="php">define( 'WP_DEBUG', 1 );
define( 'SCRIPT_DEBUG', 1 );
define( 'WP_DEBUG', 0 ); // Noncompliant, redefines constant defined 2 lines above
echo WP_DEBUG; // output "1"</code></pre>
</div>
</div>
</div>
</div>
</div>