<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Caller information attributes (<code>CallerFilePathAttribute</code>, <code>CallerLineNumberAttribute</code>, <code>CallerMemberNameAttribute</code>, and <code>CallerArgumentExpressionAttribute</code>) provide a way to get information about the caller of a method through optional parameters. But they only work right if their values aren&#8217;t provided explicitly. So if you define a method with caller info attributes in the middle of the parameter list, you put your callers in a bad position: they are forced to use named arguments if they want to use the method properly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void TraceMessage([CallerMemberName] string memberName = "",
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0,
  string message = null)  // Noncompliant
{
  /* ... */
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">void TraceMessage(string message = null,
  [CallerMemberName] string memberName = "",
  [CallerFilePath] string filePath = "",
  [CallerLineNumber] int lineNumber = 0)
{
  /* ... */
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Move "xxx" to the end of the parameter list.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_on_18_nov_2015_100125_tamas_vajk_wrote">on 18 Nov 2015, 10:01:25 Tamas Vajk wrote:</h3>
<div class="paragraph">
<p>\[~ann.campbell.2] I modified the sample code: there can&#8217;t be any parameter without a default value after parameters with default values. (So I added the <code>null</code> default value to the <code>message</code>)</p>
</div>
<div class="paragraph">
<p>And also modified the last sentence. It is possible to not specify all parameters in a call, but then you force the callers of this method to use named arguments.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_18_nov_2015_195528_ann_campbell_wrote">on 18 Nov 2015, 19:55:28 Ann Campbell wrote:</h3>
<div class="paragraph">
<p>Okay, thanks [~tamas.vajk]</p>
</div>
</div>
</div>
</div>