<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The <code>typeof</code> operator returns a string indicating the type of its argument, and the set of returned values is limited:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>"undefined"</p>
</li>
<li>
<p>"boolean"</p>
</li>
<li>
<p>"number"</p>
</li>
<li>
<p>"string"</p>
</li>
<li>
<p>"symbol" (since ECMAScript 2015)</p>
</li>
<li>
<p>"function"</p>
</li>
<li>
<p>"object" (for <code>null</code> and any other object)</p>
</li>
<li>
<p>"BigInt" (since ECMAScript 2020)</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>Compare a <code>typeof</code> expression to anything else, and the result is predefined: <code>false</code>.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function someFunc(x) {
  return typeof x === "Number"; // Noncompliant, function will always return 'false'
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function someFunc(x) {
  return typeof x === "number";
}</code></pre>
</div>
</div>
</div>
</div>
</div>