<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Using an <code>ORDER BY</code> clause in a <code>SELECT</code> helps ensure runtime reproducibility. Without one, <em>some</em> databases <em>may</em> return results in a consistent order, but it is likely to be one that appears random to users and is not optimal for processing the data.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT model   -- Noncompliant
FROM car
WHERE wheel_count = 4</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">SELECT model
FROM car
WHERE wheel_count = 4
ORDER BY year, make</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Add an "ORDER BY" clause to this select.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s1592">is related to: <a data-rspec-id="S1592" class="rspec-auto-link">S1592</a></h3>

</div>
<div class="sect2">
<h3 id="_is_related_to_s2241">is related to: <a data-rspec-id="S2241" class="rspec-auto-link">S2241</a></h3>

</div>
<div class="sect2">
<h3 id="_on_14_jul_2015_145527_pierre_yves_nicolas_wrote">on 14 Jul 2015, 14:55:27 Pierre-Yves Nicolas wrote:</h3>
<div class="paragraph">
<p>Looks good.</p>
</div>
</div>
</div>
</div>