<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>In JavaScript, every value can be coerced into a boolean value: either <code>true</code> or <code>false</code>.</p>
</div>
<div class="paragraph">
<p>Values that are coerced into <code>true</code> are said to be <em>truthy</em>, and those coerced into <code>false</code> are said to be <em>falsy</em>.</p>
</div>
<div class="paragraph">
<p>A value&#8217;s truthiness matters, and depending on the context, it can be necessary or redundant to cast a value to boolean explicitly.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>A boolean cast via double negation (<code>!!</code>) or a <code>Boolean</code> call is redundant when used as a condition, though. The condition can be written without the extra cast and behave exactly the same.</p>
</div>
<div class="paragraph">
<p>The reason is that JavaScript uses type coercion and automatically converts values to booleans in a specific situation known as a boolean context. The boolean context can be any conditional expression or statement.</p>
</div>
<div class="paragraph">
<p>For example, these <code>if</code> statements are equivalent:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">if (!!foo) {
    // ...
}

if (Boolean(foo)) {
    // ...
}

if (foo) {
    // ...
}</code></pre>
</div>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>A redundant boolean cast affects code readability. Not only the condition becomes more verbose but it also misleads the reader who might question the intent behind the extra cast.</p>
</div>
<div class="paragraph">
<p>The more concise the condition, the more readable the code.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The fix for this issue is straightforward. One just needs to remove the extra boolean cast.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">if (!!foo) {
    // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">if (foo) {
    // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_noncompliant_code_example_2">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">while (Boolean(foo)) {
    // ...
}</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution_2">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">while (foo) {
    // ...
}</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion">MDN Boolean coercion</a></p>
</li>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion">MDN Type coercion</a></p>
</li>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Glossary/Truthy">MDN Truthy</a></p>
</li>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Glossary/Falsy">MDN Falsy</a></p>
</li>
</ul>
</div>
</div>
<div class="sect2">
<h3 id="_articles_blog_posts">Articles &amp; blog posts</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://blog.alexdevero.com/truthy-falsy-values-in-javascript/">Alex Devero, How Truthy and Falsy Values in JavaScript Work</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Disallow unnecessary boolean casts.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments and links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>