<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When declaring a Windows Communication Foundation (WCF) <code>OperationContract</code> method one-way, that service method won&#8217;t return any result, not even an underlying empty confirmation message. These are fire-and-forget methods that are useful in event-like communication. Specifying a return type therefore does not make sense.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;ServiceContract&gt;
Interface IMyService
    &lt;OperationContract(IsOneWay:=True)&gt;
    Function SomethingHappened(ByVal parameter As Integer) As Integer ' Noncompliant
End Interface</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-vbnet" data-lang="vbnet">&lt;ServiceContract&gt;
Interface IMyService
    &lt;OperationContract(IsOneWay:=True)&gt;
    Sub SomethingHappened(ByVal parameter As Integer)
End Interface</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="paragraph">
<p>The rule doesn&#8217;t report if <code>OperationContractAttribute.AsyncPattern</code> is set to <code>true</code>.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>This method can&#8217;t return any values because it is marked as one-way operation.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>return type</p>
</div>
</div>
</div>
</div>