<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Numbers in JavaScript are stored as 64-bit floating-point. However, numbers that can&#8217;t be represented in such a format get rounded before assigning them to a variable.</p>
</div>
<div class="sect2">
<h3 id="_what_is_the_potential_impact">What is the potential impact?</h3>
<div class="paragraph">
<p>When you are saving a number that is too big or contains a fraction that is not expressible in the 64-bit floating-point format, it gets rounded which will lead to unexpected behaviours when it will be used.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_how_to_fix_it">How to fix it</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When you need to store a large number, use <code>BigInt()</code>. Pay attention that operations between <code>BigInt</code> and <code>Number</code> types raise an error, so you will have to adapt your code accordingly.</p>
</div>
<div class="paragraph">
<p>For numbers with digits after the decimal, it is recommended to use a library to ensure that calculation errors are not introduced by rounding.</p>
</div>
<div class="sect2">
<h3 id="_code_examples">Code examples</h3>
<div class="sect3">
<h4 id="_noncompliant_code_example">Noncompliant code example</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const foo = 2312123211345545367
const bar = 0.123456789123456789</code></pre>
</div>
</div>
</div>
<div class="sect3">
<h4 id="_compliant_solution">Compliant solution</h4>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">const foo = BigInt(2312123211345545367)
// use a library like decimal.js for storing numbers containing decimal digits</code></pre>
</div>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_how_does_this_work">How does this work?</h3>
<div class="paragraph">
<p>JavaScript numbers are stored in a double-precision 64-bit binary format IEEE 754. There are limits to what can be stored in this format both in huge numbers as in numbers with decimal digits.
When attempting to store a number that can not be represented in this format into a variable, the number gets rounded before storage.</p>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="sect2">
<h3 id="_documentation">Documentation</h3>
<div class="ulist">
<ul>
<li>
<p><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number">MDN Number</a></p>
</li>
<li>
<p><a href="https://en.wikipedia.org/wiki/Double-precision_floating-point_format">Wikipedia Double-precision floating-point format</a></p>
</li>
<li>
<p><a href="https://en.wikipedia.org/wiki/IEEE_754">Wikipedia IEEE 754 Standard</a></p>
</li>
</ul>
</div>
</div>
</div>
</div>