<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When writing UI tests, it is often required to access the UI object to assess its behavior is as expected. One can do it by name, placeholder, or some other property that can be localized. The "localized property" way is not reliable as it can change according to the language of the user.</p>
</div>
<div class="paragraph">
<p>In order to avoid to have to often update the UI tests (when the name, placeholder, &#8230;&#8203; is changing) it&#8217;s more convenient to refer to it through an "accessibilityIdentifier".</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text"></code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text"></code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://www.swiftbysundell.com/posts/getting-started-with-xcode-ui-testing-in-swift">Getting started with Xcode UI testing in Swift - Verifying state in a UI test</a></p>
</li>
</ul>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Feed the "accessibilityIdentifier" field.</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="paragraph">
<p>The UI component variable</p>
</div>
</div>
</div>
</div>