<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There&#8217;s no need to end a <code>void</code> method with a <code>return</code> statement. Since nothing is actually being returned, the <code>return</code> can be omitted; <code>void</code> will be returned implicitly.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void incrementCount() {
  this.count++;
  return;  // Noncompliant
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">public void incrementCount() {
  this.count++;
}</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Remove this useless "return" statement.</p>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_relates_to_s3626">relates to: <a data-rspec-id="S3626" class="rspec-auto-link">S3626</a></h3>

</div>
<div class="sect2">
<h3 id="_on_13_oct_2015_073909_nicolas_peru_wrote">on 13 Oct 2015, 07:39:09 Nicolas Peru wrote:</h3>
<div class="paragraph">
<p>fine by me.</p>
</div>
</div>
<div class="sect2">
<h3 id="_on_9_jan_2018_162937_amaury_levé_wrote">on 9 Jan 2018, 16:29:37 Amaury Levé wrote:</h3>
<div class="paragraph">
<p>Implemented in <a data-rspec-id="S3626" class="rspec-auto-link">RSPEC-3626</a> for C#</p>
</div>
</div>
</div>
</div>