<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>React&#8217;s useState hook setter function should not be called directly in the body of a component, otherwise it would produce an infinite render loop. This can happen by mistake. Most commonly, the setter function is called from an event handler.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { useState } from "react";

function ShowLanguage() {
    const [language, setLanguage] = useState("fr-FR");

    // Makes an infinite loop
    setLanguage(navigator.language);

    return (
      &lt;section&gt;
        &lt;h1&gt;Your language is {language}!&lt;/h1&gt;
        &lt;button onClick={() =&gt; setLanguage("fr-FR")}&gt;Je préfère le Français&lt;/button&gt;
      &lt;/section&gt;
    );
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { useState } from "react";

function ShowLanguage() {
    const [language, setLanguage] = useState(navigator.language);

    return (
      &lt;section&gt;
        &lt;h1&gt;Your language is {language}!&lt;/h1&gt;
        &lt;button onClick={() =&gt; setLanguage("fr-FR")}&gt;Je préfère le Français&lt;/button&gt;
      &lt;/section&gt;
    );
}</code></pre>
</div>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_resources">Resources</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://reactjs.org/docs/hooks-state.html">hooks-state</a> - React API reference</p>
</li>
</ul>
</div>
</div>
</div>