<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">

</div>
</div>
<div class="sect1">
<h2 id="_why_is_this_an_issue">Why is this an issue?</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When two functions have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may be confusing to maintainers. In the latter case, the code should be refactored.</p>
</div>
<div class="sect2">
<h3 id="_noncompliant_code_example">Noncompliant code example</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function calculateCode() {
  doTheThing();
  doOtherThing();
  return code;
}

function getName() {  // Noncompliant
  doTheThing();
  doOtherThing();
  return code;
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_compliant_solution">Compliant solution</h3>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">function calculateCode() {
  doTheThing();
  doOtherThing();
  return code;
}

function getName() {
  return calculateCode();
}</code></pre>
</div>
</div>
</div>
<div class="sect2">
<h3 id="_exceptions">Exceptions</h3>
<div class="ulist">
<ul>
<li>
<p>Functions with fewer than 3 lines are ignored.</p>
</li>
<li>
<p>This rule does not apply to function expressions and arrow functions because they don&#8217;t have explicit names and are often used in a way where refactoring is not applicable.</p>
</li>
</ul>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">list.map((item) =&gt; ({
  name: item.name,
  address: item.address,
  country: item.country
}));</code></pre>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_message">Message</h3>
<div class="paragraph">
<p>Update this method so that its implementation is not identical to "xxx".</p>
</div>
</div>
<div class="sect2">
<h3 id="_highlighting">Highlighting</h3>
<div class="ulist">
<ul>
<li>
<p>primary: method signature</p>
</li>
<li>
<p>secondary: original method</p>
<div class="ulist">
<ul>
<li>
<p>message: "Original implementation."</p>
</li>
</ul>
</div>
</li>
</ul>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_duplicated_by_s3502">is duplicated by: <a data-rspec-id="S3502" class="rspec-auto-link">S3502</a></h3>

</div>
<div class="sect2">
<h3 id="_on_6_oct_2017_144358_jean_christophe_collet_wrote">on 6 Oct 2017, 14:43:58 Jean-Christophe Collet wrote:</h3>
<div class="paragraph">
<p><a href="http://www.viva64.com/en/b/0363/#ID0EVHBI" class="bare">http://www.viva64.com/en/b/0363/#ID0EVHBI</a></p>
</div>
</div>
</div>
</div>